### DISCUSSION TITLE
Suitable for small custom map area?

### DISCUSSION THREAD

**Discussion by fusionconsulting** [fleaflet/flutter_map#1954] [Q&A]
I am looking into options for mapping a small area (2km x 2km) in a specific location. I have map tiles for the area starting at a specific lat/lng coordinate. Would this library be suitable for displaying this map? Can I set the origin and use custom x/y map indexes based on distance from the origin or this small map? And is it possible to restrict the map bounds to this specific region so it cannot be scrolled beyond those bounds?

**Comment by josxha** [comment]
> I am looking into options for mapping a small area (2km x 2km) in a specific location. I have map tiles for the area starting at a specific lat/lng coordinate. Would this library be suitable for displaying this map?

Yes this is possible with flutter_map. Depending on the amount of tiles you have you can either use the [AssetTileProvider](https://docs.fleaflet.dev/layers/tile-layer/tile-providers#assettileprovider). If you have a larger amount of tiles you can either use the commonly used MBTiles format or if you need to support web the PMTiles format.

> Can I set the origin and use custom x/y map indexes based on distance from the origin or this small map?

Use initCenter to set the center position. Not sure what you mean by custom x/y map indexes. 

> And is it possible to restrict the map bounds to this specific region so it cannot be scrolled beyond those bounds?

Yes, use the cameraConstraint property in the MapOptions. See https://docs.fleaflet.dev/usage/options

**Comment by fusionconsulting** [comment]
Thanks, this looks promising. 

My second question was badly worded, my coordinate system has 0,0 at the bottom left. How would I flip the Y axis to handle this. I'm looking at the examples and I think I would be able to do this in a custom CRS but not sure exactly how.

**Reply by josxha** [reply]
Try to use the TMS flag. https://docs.fleaflet.dev/getting-started/explanation#tile-basics

**Reply by JaffaKetchup** [reply]
(There's a possibility TMS is slightly broken at the moment, and I can't remember if there's a workaround or not)

**Reply by josxha** [reply]
tracking issue: https://github.com/fleaflet/flutter_map/issues/1394
